openerp/tools/mail.py

```diff
@@ -123,6 +130,30 @@ def html_sanitize(src, silent=True, strict=False, strip_style=False, strip_class
     return cleaned
 
 
+_style_whitelist = ['width', 'height', 'font-size']
+_style_replace = {
+    ''
+}
+
+def sanitize_attributes(html_root):
+    for node in html_root.iter():
+        attributes = node.attrib
+        styling = attributes.get('style')
+        valid_styles = {}
+        if styling:
+            styles = style_re.findall(styling)
+            styles = [remove_space_re.sub('', style) for style in styles]
+            for style in styles:
+                if ':' not in style:
+                    continue
+                splitted = style.split(':', 1)
+                if splitted[0] not in _style_whitelist:
+                    continue
+                valid_styles[splitted[0]] = splitted[1]
+        node.attrib['style'] = '; '.join('%s: %s' % (key, val) for (key, val) in valid_styles.iteritems())
```

Comment:
This might be worth pulling in cssutils (it's a debian stable dependency so…) it implements [CSSOM](http://www.w3.org/TR/cssom/) so the interface is not the cleanest ever, but it does have a [parseStyle](http://pythonhosted.org/cssutils/docs/parse.html#parsestyle) function to parse a `@style` declaration, manipulate it and convert it back.